### PR TITLE
Porting existing MMI section to new page

### LINKED ISSUE #22745: Confirmation redesign - port MMI warning to new designs
Currently we have these MMI warnings implemented in signature request pages, we need to discuss with MMI team and design team and appropriately port them to new designs.

cc @bschorchit , @SayaGT redesi

### PR DESCRIPTION
**Pull request by jpuri**
## **Description**

Porting existing MMI section to new page. It does not include adding test coverage or storybook - as this code is owned by MMI team - PR is not adding new code but only moving copying exiting code over to new page.

## **Related issues**

Fixes: MetaMask/metamask-extension#22745

## **Manual testing steps**
NA

### **Before**
Screenshot 2024-02-05 at 11 58 36 AM

### **After**
Screenshot 2024-02-05 at 11 55 14 AM

## **Pre-merge author checklist**

- [X] I’ve followed [MetaMask Coding Standards](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/CODING_GUIDELINES.md).
- [X] I've clearly explained what problem this PR is solving and how it is solved.
- [X] I've linked related issues
- [ ] I've included manual testing steps
- [ ] I've included screenshots/recordings if applicable
- [ ] I’ve included tests if applicable
- [ ] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [X] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.
- [X] I’ve properly set the pull request status:
  - [ ] In case it's not yet "ready for review", I've set it to "draft".
  - [X] In case it's "ready for review", I've changed it from "draft" to "non-draft".

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

**Review by segun — APPROVED** [review]

**Review by segun — DISMISSED** [review]

**Review by digiwand — CHANGES_REQUESTED** [review]
interesting this component wasn't added to Sign-in With Ethereum (SIWE). This might have been missed when SIWE was implemented.

requesting changes for file path name, classname and action item to port over CSS

File: ui/pages/confirmations/components/confirm/mmi-signature-section/mmi-signature-section.tsx [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+    selectedAccount &&
+    fromAccount &&
+    selectedAccount.address === fromAccount.address
+  ) {
+    return null;
+  }
+
+  return (
+    <Box
+      className="request-signature__mismatch-info"
```
[the remark this answers is not in the document]
Comment (jpuri) [reply ui/pages/confirmations/components/confirm/mmi-signature-section/mmi-signature-section.tsx:60]: @digiwand : these styles are added by MMI team and this I will not be able to change these classes.

File: ui/pages/confirmations/components/confirm/mmi-signature-section/mmi-signature-section.tsx [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+  shortenAddress,
+} from '../../../../../helpers/utils/util';
+import { useI18nContext } from '../../../../../hooks/useI18nContext';
+import {
+  accountsWithSendEtherInfoSelector,
+  currentConfirmationSelector,
+  getSelectedInternalAccount,
+} from '../../../../../selectors';
+
+const MMISignatureSection: React.FC = memo(() => {
```
[the remark this answers is not in the document]
Comment (jpuri) [reply ui/pages/confirmations/components/confirm/mmi-signature-section/mmi-signature-section.tsx]: I renamed the component to `MMISignatureMismatchBanner`

File: ui/components/app/confirm/mmi-signature-section/mmi-signature-section.tsx [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+} from '../../../../helpers/utils/util';
+import { useI18nContext } from '../../../../hooks/useI18nContext';
+import {
+  accountsWithSendEtherInfoSelector,
+  currentConfirmationSelector,
+  getSelectedInternalAccount,
+} from '../../../../selectors';
+import { Box, Icon, IconName, Text } from '../../../component-library';
+
+const MMISignatureMismatchBanner: React.FC = memo(() => {
```
Comment (cryptotavares) [inline ui/components/app/confirm/mmi-signature-section/mmi-signature-section.tsx:25]: Seems to be a duplicate and not being used anywhere... Is there a reason for adding this?

Comment (jpuri) [reply ui/components/app/confirm/mmi-signature-section/mmi-signature-section.tsx:25]: Oh that is merge issue, let me delete this.

File: ui/pages/confirmations/components/confirm/mmi-signature-mismatch-banner/mmi-signature-mismatch-banner.tsx [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+const MMISignatureMismatchBanner: React.FC = memo(() => {
+  const t = useI18nContext();
+  const currentConfirmation = useSelector(currentConfirmationSelector);
+  const selectedAccount = useSelector(getSelectedInternalAccount);
+  const allAccounts = useSelector(accountsWithSendEtherInfoSelector);
+
+  const fromAccount = useMemo(() => {
+    if (
+      !currentConfirmation ||
+      currentConfirmation.type !== MESSAGE_TYPE.PERSONAL_SIGN ||
```
Comment (pedronfigueiredo) [inline ui/pages/confirmations/components/confirm/mmi-signature-mismatch-banner/mmi-signature-mismatch-banner.tsx:39]: Just double-checking if this component is not shown for other signature types?

```js
  ETH_SIGN: 'eth_sign',
  ETH_SIGN_TYPED_DATA: 'eth_signTypedData',
  ETH_SIGN_TYPED_DATA_V3: 'eth_signTypedData_v3',
  ETH_SIGN_TYPED_DATA_V4: 'eth_signTypedData_v4',
```

Comment (jpuri) [reply ui/pages/confirmations/components/confirm/mmi-signature-mismatch-banner/mmi-signature-mismatch-banner.tsx:39]: I found it there in personal sign, as we port other signatures to new designs we can enable this for other signatures also if they support it.

**Review by pedronfigueiredo — DISMISSED** [review]

**Review by blackdevelopa — DISMISSED** [review]

**Review by pedronfigueiredo — APPROVED** [review]

**Review by blackdevelopa — APPROVED** [review]

**Review by digiwand — APPROVED** [review]
Lgtm! 

documentation note:
The code logic in this PR was extracted from `ui/pages/confirmations/components/signature-request-original/signature-request-original.component.js`

**Pull request merged**